ipa-core/src/net/test.rs

```diff
@@ -404,3 +686,48 @@ const TEST_HPKE_PUBLIC_KEY: &str = "\
 const TEST_HPKE_PRIVATE_KEY: &str = "\
 a0778c3e9960576cbef4312a3b7ca34137880fd588c11047bd8b6a8b70b5a151
 ";
+
+#[cfg(all(test, unit_test))]
+mod tests {
+    use super::TestConfigBuilder;
+    use crate::{helpers::HelperIdentity, net::test::Ports};
+
+    /// This simple test makes sure that testing networks are created properly.
+    /// The network itself won't be excersized as that's tested elsewhere.
+    #[test]
+    fn create_4_shard_http_network() {
+        let ports: Vec<Ports<Vec<u16>>> = vec![
```

Comment:
do you want to test that overlapping ports are rejected?

Reply:
Sure, I can add that